src/core/reference/include/openvino/reference/reshape.hpp

```diff
@@ -23,6 +23,19 @@ inline void reshape(const char* in, char* out, const Shape& in_shape, size_t ele
     std::memcpy(out, in, shape_size(in_shape) * elem_size);
 }
 
+/**
+ * @brief Basic reshape operation for string element type.
+ *        Input data are simply copied to the output.
+ *        Target shape should be set on the Tensor.
+ *
+ * @param in         Pointer to input data.
+ * @param out        Pointer to output data.
+ * @param in_shape   Input data shape.
+ */
+inline void reshape(const std::string* in, std::string* out, const Shape& in_shape) {
+    std::copy_n(in, shape_size(in_shape), out);
+}
+
```

Comment:
Let us enable CPU fallback

Reply:
As Reshape operation is basically just a copy of the data, and in CPU plugin it's common for Reshape/Squeeze/Unsqueeze, so I have proposed Reshape CPU implementation for string instead of fallback.
I will ask for review CPU plugin maintainers.

Reply:
So I've decided to revert CPU changes from this PR, I will create another one for CPU plugin to review them separately.
Let's the merge reference first.